### DISCUSSION TITLE
Group layout + `strokeWidth`

### DISCUSSION THREAD

**Discussion by wuxingping** [fabricjs/fabric.js#8693] [General]
version:6.0.0-beta1
description:something is wrong as follow:
https://codesandbox.io/s/busy-water-wmh0xp?file=/src/App.vue
![image](https://user-images.githubusercontent.com/41353266/218678000-087883bc-4978-4a04-8479-ed2172f75738.png)

**Comment by wuxingping** [comment]
it works well.i overwrite onLayout,as follow:
`onLayout(context, result){
          this.set({ width: result.width-this.strokeWidth, height: result.height-this.strokeWidth });
     }`

**Comment by ShaMan123** [comment]
It is exactly what I wanted to suggest
https://codesandbox.io/s/friendly-fermi-3vy0tw?file=/src/container.ts

**Reply by wuxingping** [reply]
> It is exactly what I wanted to suggest https://codesandbox.io/s/friendly-fermi-3vy0tw?file=/src/container.ts

by the way,how to import these types.as follow:
![image](https://user-images.githubusercontent.com/41353266/218723169-e165feb1-409e-47ce-9f2f-a78dc8e6e96a.png)

**Reply by ShaMan123** [reply]
I am not sure... It is a new thing that is related to the build system
Try using autocomplete (ctrl + shift on windows)

**Reply by wuxingping** [reply]
> I am not sure... It is a new thing that is related to the build system Try using autocomplete (ctrl + shift on windows)
i tried.but there is a tip as follow:
![image](https://user-images.githubusercontent.com/41353266/218899513-6d24a72a-c53c-48b1-ae09-88e7fd1e9ce6.png)

**Reply by wuxingping** [reply]
> > I am not sure... It is a new thing that is related to the build system Try using autocomplete (ctrl + shift on windows)
> > i tried.but there is a tip as follow:
> > ![image](https://user-images.githubusercontent.com/41353266/218899513-6d24a72a-c53c-48b1-ae09-88e7fd1e9ce6.png)

i guess because of the path like as from 'fabric..'  point to "d:/znqc_piccel/node_modules/fabric/dist/index",so the path "fabric/dist/src/shapes/Group" is inoperative.

**Comment by ShaMan123** [comment]
try giving a title to the discussion that is meaningful
see how I changed it

**Reply by wuxingping** [reply]
> try giving a title to the discussion that is meaningful see how I changed it

okay,thanks a lot

**Reply by wuxingping** [reply]
by the way,i want to ask where is it the object inside group when i drag it out group．if i want it display when it dragged out completely group as follow．how should  i  do?　
![image](https://user-images.githubusercontent.com/41353266/218726693-43651a07-84cb-432c-8454-bdcf551e888e.png)

**Reply by ShaMan123** [reply]
It is there but it is clipped by the clip path

**Reply by ShaMan123** [reply]
You can `triggerLayout({ layout: 'fit-content' })` and see for yourself

**Reply by ShaMan123** [reply]
I would remove the clip path while dragging
If that is not good enough because you need to clip the other objects then look into Canvas#_chooseObjectsToRender and try:
```
_chooseObjectsToRender() {
   return [...this._objects, this._activeObject]
}
```

But I am sure you will encounter edge cases.
I need to solve this.
